packages/@atjson/renderer-html/src/index.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  Code,
+  Heading,
+  Image,
+  Link,
+  List
+} from '@atjson/offset-annotations';
+import Renderer from '@atjson/renderer-hir';
+
+export default class HTMLRenderer extends Renderer {
+
+  /**
+   * Renders an HTML string from an object.
+   *
+   * To invoke this, use `yield* this.$()` to get the output.
+   * This will forward the children to this function to get wrapped
+   * in the HTML tag.
+   *
+   * @param tagName The HTML tag name to use.
+   * @param props The HTML attributes (if there are any) and whether the element is self-closing.
+   */
+  *$(tagName: string, props?: { attributes?: any, selfClosing?: boolean }) {
+    let attributes = props ? props.attributes || {} : {};
+    let htmlAttributes = Object.keys(attributes).reduce((results, key) => {
+      let value = attributes[key];
+      if (typeof value === 'number') {
+        results.push(`${key}=${value}`);
+      } else if (typeof value === 'boolean' && value === true) {
+        results.push(`${key}`);
+      } else if (value != null && value !== false) {
+        results.push(`${key}="${value}"`);
+      }
+      return results;
+    }, [] as string[]);
+    let innerHTML: string[] = yield;
+
+    let selfClosing = props ? props.selfClosing : false;
+    if (selfClosing) {
+      if (htmlAttributes.length) {
+        return `<${tagName} ${htmlAttributes.join(' ')} />`;
+      }
+
+      return `<${tagName} />`;
+    }
+
+    if (htmlAttributes.length) {
+      return `<${tagName} ${htmlAttributes.join(' ')}>${innerHTML.join('')}</${tagName}>`;
+    }
+
+    return `<${tagName}>${innerHTML.join('')}</${tagName}>`;
+  }
+
+  *root() {
+    let html = yield;
+    return html.join('');
+  }
+
+  *Blockquote() {
+    return yield* this.$('blockquote');
+  }
+
+  *Bold() {
+    return yield* this.$('strong');
+  }
+
+  *Code(code: Code) {
+    let codeSnippet = yield* this.$('code');
+
+    if (code.attributes.style === 'block' || code.attributes.style === 'fence') {
+      return `<pre>${codeSnippet}</pre>`;
+    }
+    return codeSnippet;
```

Comment:
Do we need to update the HTML converter to turn `<pre><code>` => `{ type: Code }`?